### PR TITLE
fix: empty errors in serialized instrumentation data

### PR DESCRIPTION
**Pull request by PeterSchafer**
* When errors don't have an id, don't add empty error structures to the serialized data
* ensure category vector is lower case
* add GetDuration()

### CODE REVIEW

File: pkg/instrumentation/categories.go [hunk 1]
```diff
@@ -62,14 +62,14 @@ func DetermineCategoryFromArgs(args []string, knownCommands []string, flagsAllow
 			flagName := strings.TrimLeft(flagParts[0], "-")
 
 			if slices.Contains(flagsAllowList, flagName) {
-				flags = append(flags, flagName)
+				flags = append(flags, strings.ToLower(flagName))
```
Comment (cmars) [inline pkg/instrumentation/categories.go:65]: Could we add a test case for this?

File: pkg/analytics/instrumentation_collector.go [hunk 2]
```diff
@@ -228,13 +232,14 @@ func toInteractionErrors(errors []error) *[]api.InteractionError {
 
 func toInteractionError(e error) *api.InteractionError {
```
Comment (cmars) [inline pkg/analytics/instrumentation_collector.go:233]: Could we add a test case for this function?

**Review by cmars — COMMENTED** [review]
LGTM with a couple of test cases added

File: pkg/analytics/instrumentation_collector.go (hunk 2 above)
Comment (PeterSchafer) [reply pkg/analytics/instrumentation_collector.go:233]: Done!

File: pkg/instrumentation/categories.go (hunk 1 above)
Comment (PeterSchafer) [reply pkg/instrumentation/categories.go:65]: Done!

**Pull request merged**